### PR TITLE
[Add] version-automation

### PR DESCRIPTION
**Pull request by Rudra-IITM**
- Updated snapcraft.yaml to implement automated versioning for the snap.
- Streamline the inclusion of source packages in the repository pull for ghostscript, eliminating unnecessary components for this specific snap.

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -216,6 +216,11 @@ parts:
       - --enable-freetype
       - --without-tesseract
       - --datarootdir=/snap/ps-printer-app/current/usr/share/
+    override-pull: |
+      set -eux
+      # Do the actual pull task
+      craftctl default
+      rm -rf gpdl pcl xps
```
Comment (tillkamppeter) [inline snap/snapcraft.yaml:223]: For Snap it is no problem to do it this way to not build support for undesired input data formats which have extra dependencies (like TIFF), more "elegant" would be to find the correct `./configure` command line options which do the same thing.

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -543,30 +546,63 @@ parts:
...
-      # some software which is under continuous development and
-      # milestones of the development turn into releases, it is simply
-      # versioned by the date (a cron job on OpenPrinting "releases"
-      # it every day.
-      #
-      # As we want to download it from GIT and not grab a daily
-      # tarball for sake of Snap automation we simply take the date of
-      # the day after the day of the execution of the pull task of
-      # this part as the version number (this is the version in which
-      # the current GIT state would land).
```
Comment (tillkamppeter) [inline snap/snapcraft.yaml:562]: Please do not remove this comment, put it to the beginning of the version automation code.

**Comment by Rudra-IITM** [comment]
@tillkamppeter, would you mind examining the modifications, please?

**Comment by tillkamppeter** [comment]
@Rudra-IITM thanks a lot for this PR, I have merged it now.

**Pull request merged**